### PR TITLE
Fix build failure

### PR DESCRIPTION
**Pull request by hixichen**
This change fixed serveral issues:
        - the build failed when use docker image to compile darwin binary inside ubuntu image.
          Due to the imported cgo, cross compile needed to be updated. A script is added  to build binary with go installed on machine.

        - protoc inside toolchain image.

### CODE REVIEW

File: pbft/HyperMake [hunk 1]
```diff
@@ -44,18 +36,6 @@ targets:
     artifacts:
       - bin/linux/truechain-engine
 
-  build-darwin:
```
Comment (arcolife) [inline pbft/HyperMake]: so you've updated Dockerfile with xgo but removed build-darwin?

File: pbft/build.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (arcolife) [inline pbft/build.sh]: I'm guessing you forgot to remove this build script?  because, hmake build :)

File: pbft/src/vendor/manifest [hunk 3]
```diff
@@ -1243,13 +1243,30 @@
 			"path": "protobuf/field_mask",
 			"notests": true
 		},
+		{
```
Comment (arcolife) [inline pbft/src/vendor/manifest]: hey in this same commit, could we address the "dependency strip down to minimal" issue?

**Comment by miamiruby** [comment]
this is still having problems...

paulkruger@scream:~/Sites/truechain/truechain-consensus-core/pbft(devel)$ hmake build-darwin
HyperMake v1.3.1 https://github.com/evo-cloud/hmake

=> toolchain 18:02:08.551
:] toolchain
=> deps 18:02:08.553
:] deps
=> build-darwin 18:02:08.555
[build-darwin]      + export GOOS=darwin
[build-darwin]      + GOOS=darwin
[build-darwin]      + export GOARCH=amd64
[build-darwin]      + GOARCH=amd64
[build-darwin]      + protoc -I src/pbft-core/fastchain/ src/pbft-core/fastchain/fastchain.proto --go_out=plugins=grpc:src/pbft-core/fastchain/
[build-darwin]      ++ pwd
[build-darwin]      ++ pwd
[build-darwin]      + export GOPATH=/go:/src:/src:/src/..
[build-darwin]      + GOPATH=/go:/src:/src:/src/..
[build-darwin]      + OUTDIR=bin/darwin
[build-darwin]      + mkdir -p bin/darwin
[build-darwin]      + '[' darwin -eq linux ']'
[build-darwin]      ./support/scripts/build.sh: line 20: [: darwin: integer expression expected
[build-darwin]      ++ git_commit_hash
[build-darwin]      +++ git rev-parse --short HEAD
[build-darwin]      fatal: not a git repository (or any parent up to mount point /)
[build-darwin]      Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
[build-darwin]      ++ echo
[build-darwin]      + LDFLAGS='-s -w -X common.GitCommitHash='
[build-darwin]      + go build -o bin/darwin/pbft-client -ldflags '-s -w -X common.GitCommitHash=' ./src/pbft-core/client/
[build-darwin]      src/vendor/github.com/ethereum/go-ethereum/crypto/signature_cgo.go:27:2: build constraints exclude all Go files in /src/src/vendor/github.com/ethereum/go-ethereum/crypto/secp256k1
[build-darwin]      + go build -o bin/darwin/truechain-engine -ldflags '-s -w -X common.GitCommitHash=' ./src/pbft-core/pbft-sim-engine/
[build-darwin]      src/vendor/github.com/ethereum/go-ethereum/crypto/signature_cgo.go:27:2: build constraints exclude all Go files in /src/src/vendor/github.com/ethereum/go-ethereum/crypto/secp256k1
:( build-darwin 18:02:17.171 [+8.616748s] exit status 1
╔════════════╤═══════╤═════════╤════════════╤════════════╤═════════════╗
║Target      │Result │ Duration│Start       │Finish      │Error        ║
╠════════════╪═══════╪═════════╪════════════╪════════════╪═════════════╣
║toolchain   │Skipped│         │18:02:08.551│18:02:08.551│             ║
║deps        │Skipped│         │18:02:08.553│18:02:08.553│             ║
║build-darwin│Failure│8.616748s│18:02:08.555│18:02:17.171│exit status 1║
╚════════════╧═══════╧═════════╧════════════╧════════════╧═════════════╝
ERROR: exit status 1
paulkruger@scream:~/Sites/truechain/truechain-consensus-core/pbft(devel)$

File: pbft/HyperMake (hunk 1 above)
Comment (hixichen) [reply pbft/HyperMake]: No. If you check the msg in gitter, this is the quick fix.

**Comment by hixichen** [comment]
@miamiruby  this change is not merged. 

Once it is mergered, there is not 'hmake build-darwin'. 

for darwin,  just run ./build.sh

File: pbft/HyperMake (hunk 1 above)
Comment (arcolife) [reply pbft/HyperMake]: but then why are we importing xgo in Dockerfile in the same commit?

File: pbft/src/vendor/manifest (hunk 3 above)
Comment (hixichen) [reply pbft/src/vendor/manifest]: do you have any quick way to hack for gvt remove?

**Comment by miamiruby** [comment]
Cool I will test it

**Comment by miamiruby** [comment]
it compiled.... now I have 2 files in the bin

paulkruger@scream:~/Sites/truechain/truechain-consensus-core/pbft/bin/darwin(devel*)$ ./pbft-client
[.]Loading IP configs...

REQUESTS count -  10
just fetched private key for Client
<nil>
Finish sending the requests.
[.]Firing up client executioner...

Test finished. Time cost: 133.145µs

**Pull request merged**